pom.xml

```diff
@@ -345,22 +346,18 @@
                                 </goals>
                                 <phase>test</phase>
                             </execution>
-                            <execution>
-                                <id>build-native</id>
-                                <goals>
-                                    <goal>build</goal>
-                                </goals>
-                                <phase>package</phase>
-                            </execution>
                         </executions>
                         <configuration>
-                            <imageName>sqlite-jdbc-osinfo</imageName>
-                            <mainClass>org.sqlite.util.OSInfo</mainClass>
+                            <fallback>false</fallback>
+                            <verbose>true</verbose>
                             <buildArgs>
-                                <buildArg>--no-fallback</buildArg>
-                                <buildArg>--verbose</buildArg>
-                                <buildArg>--enable-url-protocols=jar</buildArg>
-                                <buildArg>--report-unsupported-elements-at-runtime</buildArg>
+                                <!--
+                                ArchUnit tests don't run in native-image tests.
+                                Remove the ArchUnit JUnit Engine from the ServiceLoader.
+                                -->
+                                <buildArg>
+                                    -H:ServiceLoaderFeatureExcludeServiceProviders=com.tngtech.archunit.junit.internal.ArchUnitTestEngine
```

Comment:
Do we need this, since we already have `<exclude>**/architecture/*.java</exclude>` below in the Surefire configuration?

Reply:
Yes, native-image auto-detects service loaders, so the `ArchUnitTestEngine` is included. At some point during native-image generation it gets instantiated which causes runtime-buildtime initialization issues. Excluding the service entirely solves the same issue you solved with this config: https://github.com/gotson/sqlite-jdbc/blob/fix-native-test/pom.xml#L318

Reply:
I'm fine with your change, we don't need to run ArchUnit tests on native anyway.